### ISSUE TITLE
Cxx as standalone llvm/clang library

### ISSUE THREAD

**Issue by mingodad** [Keno/Cxx.jl#249]
Hello !
I do wanto to use something like Cxx with https://github.com/mingodad/squilu and I started to change bootstrap.cpp to compile standalone see here https://gist.github.com/mingodad/99d34ad93f59def5ad0ee809681be0e1 it actually compile and link with with clang 3.5/3.6/3.7/3.8, but didn't run/make anything util right now.
If someone want to join efforts on it tell me !
Cheers !

**Comment by mingodad** [comment]
I did an update that try to parse and execute the "Example 1: Embedding a simple C++ function in Julia" but it's not working yet.

Keno could you give a hand on it ?

I want a c++ standalone version showing all functionality available by Cxx.jl, I looked at the julia side but the julia wrapper syntax is not familiar to me (a bit hard to follow).  

Cheers !

**Comment by Keno** [comment]
I'm not sure you'll have much success here. Much of the actual logic is on the julia side, including the execution engine. `bootstrap.cpp` is mostly a dump wrapper around clang.

**Comment by mingodad** [comment]
Have you looked at it ?

**Comment by Keno** [comment]
Yes

**Comment by mingodad** [comment]
Could you run it ?
Can you help making the parse and execution of "mycppfunction" work ?
Cheers !

**Comment by Keno** [comment]
As I said, you're missing most of the logic to actually make it work. In particular, you don't have an execution engine.

**Comment by mingodad** [comment]
Can the execution engine be implemented in a simple way ?
I do not much experience with all the internals of llvm/clang but I do have some knowledge with C++, after see your project mentioned on hacker news and looking through the bootstrap.cpp I followed the link to http://www.ibm.com/developerworks/library/os-createcompilerllvm2/index.html and could get the examples to work with the llvm/clang 3.8 and it made me look deeper on your project. And I still think that it has potential as standalone.
If I could get it to work in C++ probably I would be able to improve it later and contribute back to your project. Actually I needed to already make some changes to make it compile with clang 3.8, although you have some preprocessor macros to deal with llvm/clang 3.8 it was not actually working before and now it at least compile.

Cheers !

**Comment by Keno** [comment]
It's probably not too bad to make work, you'd need MCJIT and the ability to split the shadow module. You can see how julia does it here: https://github.com/JuliaLang/julia/blob/c070ac80ac218d7335e757e54bb5a31b8196e98d/src/ccall.cpp#L497.

**Comment by mingodad** [comment]
Also if we could make it standalone most of the code you have on the julia side could be eliminated and other projects could benefit from it.
Tank you ! I'll look at it right now !
